mantle/kola/harness.go

```diff
@@ -783,11 +783,28 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 			}
 		}
 
-		if caughtTestError {
+		allDenylisted := true
+		for name := range tests {
+			isDenylisted, err := testIsDenyListed(name)
+			if err != nil {
+				fmt.Printf("Error while checking denylist for test")
+				continue
+			}
+
+			if !isDenylisted {
+				allDenylisted = false
+				break
+			}
+		}
+
+		if allDenylisted {
+			fmt.Printf("All tests are denylisted, output in %v\n", outputDir)
```

Comment:
this still doesn't allow the process to return with success. We need to return early here but we can do better and run this check only if the suite errored with no tests to run. Something like this:

```patch
$ git diff mantle/kola/harness.go
diff --git a/mantle/kola/harness.go b/mantle/kola/harness.go
index fb5d1cf2f..4d9f4c8fe 100644
--- a/mantle/kola/harness.go
+++ b/mantle/kola/harness.go
@@ -797,8 +797,9 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
                        }
                }
 
-               if allDenylisted {
-                       fmt.Printf("All tests are denylisted, output in %v\n", outputDir)
+               if suiteErr == harness.SuiteEmpty && allDenylisted {
+                       fmt.Printf("There are no tests to run because all tests are denylisted. Output in %v\n", outputDir)
+                       return nil
                } else if caughtTestError {
                        fmt.Printf("FAIL, output in %v\n", outputDir)
                } else {
```